lib/src/io_client.dart

```diff
@@ -72,3 +73,16 @@ class IOClient extends BaseClient {
     _inner = null;
   }
 }
+
+Uri _responseUrl(Request request, HttpClientResponse response) {
```

Comment:
> K you want it static right?

No, as an instance method. I don't like making things static unless they have to be.

> Was just following along with what we we're doing with free functions in `message.dart`

Those are used in constructors, which is pretty different. Still, I might make them static methods.